File: C++/level2/멀쩡한_사각형.cpp
#include <algorithm>

using namespace std;

int findGcd(int n, int m){
    if(n < m)
        swap(n, m);

    while(m != 0){
        n = n%m;
        swap(n, m);
    }
    return n;
}

long long solution(int w, int h) {
    int gcd = findGcd(w, h);

    return (long long)w*h-(w/gcd + h/gcd -1)*gcd;
}

File: JavaScript/level2/멀쩡한_사각형.js
function gcd(a, b){
    if(a === 0)
        return b;
    return gcd(b%a, a);
}

function solution(w, h) {
    let val;
    
    if(w > h)
        val = gcd(h, w);
    else
        val = gcd(w, h);

    return w*h - (w/val + h/val - 1)*val;
}